### PR TITLE
Updated Program.cs for Templates

### LINKED ISSUE #9882: [Solution included] Unable to add the EntityFramework Core migration
### Describe the bug
When I tried to add initial Entity Framework Core migration it couldn't create the database context.

### To Reproduce
Steps to reproduce the behaviour:
1. Create a solution with name `Test`.
2. Create the web application project in the solution with name `WebApp` from the latest `dev` branch using the following commands:
```cmd
dotnet new -i "OrchardCore.ProjectTemplates::1.0.0-rc2-*" --nuget-source https://nuget.cloudsmith.io/orchardcore/preview/v3/index.json
```
```cmd
dotnet new occms
```
3. Create the .NET Core 5 library with name `Core`;
4. Add EntityFramework Core packages to both projects.
5. Create the EF Core database context `ApplicationDbContext` in the `Core` project and one entity type.
6. Try to create the initial migration. Run the following command in the solution's directory: 
```cmd
dotnet ef migrations add Initial -c ApplicationDbContext -s .\WebApp -v -p .\Core
```
7. Get the error:
```
Unable to create an object of type 'ApplicationDbContext'. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728
```

### Expected behavior
Migration will be created successfully.

### Screenshots
image

### Solution
The problem was in the `Program.cs`. I renamed the `BuildHost` static method to `CreateHostBuilder` and changed the return type to `IHostBuilder`.

Before:
`Program.cs`
```csharp
    public class Program
    {
        public static Task Main(string[] args)
            => BuildHost(args).RunAsync();

        public static IHost BuildHost(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureLogging(logging => logging.ClearProviders())
                .ConfigureWebHostDefaults(webBuilder => webBuilder
                    .UseStartup<Startup>()
                    .UseNLogWeb())
                .Build()
            ;
    }
```
After:
`Program.cs`
```csharp
    public class Program
    {
        public static Task Main(string[] args)
            => CreateHostBuilder(args).Build().RunAsync();

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureLogging(logging => logging.ClearProviders())
                .ConfigureWebHostDefaults(webBuilder => webBuilder
                    .UseStartup<Startup>()
                    .UseNLogWeb())
            ;
    }
```

### PR DESCRIPTION
**Pull request by ns8482e**
Fixes #9882

Allow EF to use App settings

### CODE REVIEW

**Comment by ns8482e** [comment]
It would be nice to include `launch.json` and `task.json` in template for VS Code

**Comment by sebastienros** [comment]
Should we also change the main Cms project? And also OrchardCore.Samples?

**Comment by ns8482e** [comment]
> Should we also change the main Cms project? And also OrchardCore.Samples?

roger that

**Comment by Skrypt** [comment]
This breaks automatic startup of the website in Visual Studio Code.

File: src/OrchardCore.Cms.Web/Program.cs [hunk 1]
```diff
@@ -8,16 +8,18 @@ namespace OrchardCore.Cms.Web
...
-            => BuildHost(args).RunAsync();
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
 
-        public static IHost BuildHost(string[] args) =>
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
                 .ConfigureLogging(logging => logging.ClearProviders())
```
Comment (Skrypt) [inline src/OrchardCore.Cms.Web/Program.cs:18]: This breaks automatic startup of the website in Visual Studio Code.

File: src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs [hunk 2]
```diff
@@ -14,10 +14,12 @@ namespace OrchardCore.Templates.Cms.Web
...
-            => BuildHost(args).RunAsync();
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
 
-        public static IHost BuildHost(string[] args) =>
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
             Host.CreateDefaultBuilder(args)
                 .ConfigureLogging(logging => logging.ClearProviders())
```
Comment (Skrypt) [inline src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs:24]: This breaks automatic startup of the website in Visual Studio Code.

Comment (ns8482e) [reply src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs:24]: @Skrypt  not sure what you mean by automatic startup?  the Launch task works well with F5

Comment (Skrypt) [reply src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs:24]: It should open up your default browser automatically in the end. When I comment out this line it does work back.

**Comment by Skrypt** [comment]
You can also remove the firstlline : using System.Threading.Tasks;

File: src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs (hunk 2 above)
Comment (ns8482e) [reply src/Templates/OrchardCore.ProjectTemplates/content/OrchardCore.Templates.Cms.Web/Program.cs:24]: updated to `launch.json` to use `serverReadyAction`

File: src/OrchardCore.Cms.Web/Program.cs (hunk 1 above)
Comment (ns8482e) [reply src/OrchardCore.Cms.Web/Program.cs:18]: updated to `launch.json` to use `serverReadyAction`

**Comment by Skrypt** [comment]
serverReadyAction 👍🏼 
Though not fixing the issue with ConfigureLogging. Maybe it's removing too much logging providers that VS Code debugger relies on.

**Comment by ns8482e** [comment]
> serverReadyAction 👍🏼
> Though not fixing the issue with ConfigureLogging. Maybe it's removing too much logging providers that VS Code debugger (omnisharp) relies on.

Changing to `serverReadyAction` worked with `NLog`

**Comment by ns8482e** [comment]
> You can also remove the firstlline : using System.Threading.Tasks;

Removed.

**Comment by Skrypt** [comment]
I just want to make sure browser still gets started automatically. Will test this with my own website to make sure, else it looks fine.

**Review by Skrypt — APPROVED** [review]

**Comment by Skrypt** [comment]
https://github.com/OrchardCMS/OrchardCore/blob/ce53b9a90e2ba329e9b0a039a298d489302edefe/src/OrchardCore.Cms.Web/Properties/launchSettings.json#L25

https needs to be before http there too

**Pull request merged**